src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -96,7 +106,15 @@ def create_common_request_options(url, headers, options)
 
     # Ensure multiple requests are not made on the same connection
     headers["Connection"] = "close"
-    request_options = RequestOptions.new(build_url(url))
+
+    # These defaults match what the agent attaches to its HTTP requests
+    headers["X-Puppet-Version"] ||= Puppet.version
+    headers["User-Agent"] ||= Puppet[:http_user_agent]
```

Comment:
The main thing with the user agent string is that it's logged in HTTP server access logs and having unique user agents can be helpful to understand who made the request. If puppet apply and puppetserver both connected to puppetdb, then it might be useful to have different user agent strings. Puppet sends a user agent string of the form:

```
User-Agent: Puppet/7.0.0 Ruby/2.5.7-p206 (x86_64-darwin18)
```

So something like this might be useful (using  puppetserver's version, `JRUBY_VERSION` and `PLATFORM`):

```
User-Agent: Puppetserver/7.0.0 JRuby/9.2.6.0 (jruby)
```

But this is super lower priority. Also t appears `Puppet[:http_user_agent]` is already "different" due to the `java` platform:

```
 "Puppet/7.0.0 Ruby/2.5.3-p0 (java)"
```

Reply:
Given that getting Puppet Server's version into the Ruby code is hard (I looked into it a bit), I think leaving this as-is, with the `(java)` identifier, is okay for now. Still an improvement over sending nothing before. I think I'll file a follow-up ticket so we can look into plumbing that through and cleaning this up.

Reply:
Added a comment about this to the code too.